### PR TITLE
Add basic Instrumentation tests to replace robo tests

### PR DESCRIPTION
**Pull request by danesfeder**
This PR adds instrumentation tests that aim to replace the robo-testing we currently use in Firebase that is failing because of a `Gson` `NoSuchMethodError `:

```
java.lang.NoSuchMethodError: No direct method <init>(Lcom/google/gson/Gson;)V in class 
Lcom/google/gson/Gson$1; or its super classes (declaration of 'com.google.gson.Gson$1' appears in /data/app/com.google.android.apps.mtaas.crawler-1/base.apk)
	at com.google.gson.Gson.doubleAdapter(Gson.java:278)
	at com.google.gson.Gson.<init>(Gson.java:217)
	at com.google.gson.GsonBuilder.create(GsonBuilder.java:572)
	at com.mapbox.android.telemetry.TelemetryClient.sendBatch(TelemetryClient.java:69)
	at com.mapbox.android.telemetry.TelemetryClient.sendEvents(TelemetryClient.java:50)
	at com.mapbox.android.telemetry.MapboxTelemetry.sendEvents(MapboxTelemetry.java:372)
	at com.mapbox.android.telemetry.MapboxTelemetry.sendEventsIfPossible(MapboxTelemetry.java:351)
	at com.mapbox.android.telemetry.MapboxTelemetry.sendEventIfWhitelisted(MapboxTelemetry.java:458)
	at com.mapbox.android.telemetry.MapboxTelemetry.push(MapboxTelemetry.java:133)
	at com.mapbox.mapboxsdk.maps.MapView.onCreate(MapView.java:277)
	at com.mapbox.services.android.navigation.testapp.activity.MockNavigationActivity.onCreate(MockNavigationActivity.java:89)
	at android.app.Activity.performCreate(Activity.java:5990)
	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1106)
	at androidx.test.runner.MonitoringInstrumentation.callActivityOnCreate(MonitoringInstrumentation.java:215)
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2278)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2387)
	at android.app.ActivityThread.access$800(ActivityThread.java:151)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1303)
	at android.os.Handler.dispatchMessage(Handler.java:102)
	at androidx.test.espresso.base.Interrogator.a(Interrogator.java:19)
	at androidx.test.espresso.base.UiControllerImpl.a(UiControllerImpl.java:142)
	at androidx.test.espresso.base.UiControllerImpl.a(UiControllerImpl.java:134)
	at androidx.test.espresso.base.UiControllerImpl.a(UiControllerImpl.java:34)
	at androidx.test.espresso.action.MotionEvents.a(MotionEvents.java:74)
	at androidx.test.espresso.action.MotionEvents.a(MotionEvents.java:52)
	at androidx.test.espresso.action.Tap.c(Tap.java:9)
	at androidx.test.espresso.action.Tap.a(Tap.java:19)
	at androidx.test.espresso.action.Tap$1.b(Tap.java:2)
	at androidx.test.espresso.action.GeneralClickAction.perform(GeneralClickAction.java:22)
	at androidx.test.espresso.ViewInteraction$SingleExecutionViewAction.perform(ViewInteraction.java:9)
	at androidx.test.espresso.ViewInteraction.a(ViewInteraction.java:78)
	at androidx.test.espresso.ViewInteraction.a(ViewInteraction.java:94)
	at androidx.test.espresso.ViewInteraction$1.call(ViewInteraction.java:3)
	at java.util.concurrent.FutureTask.run(FutureTask.java:237)
	at android.os.Handler.handleCallback(Handler.java:739)
	at android.os.Handler.dispatchMessage(Handler.java:95)
	at android.os.Looper.loop(Looper.java:135)
	at android.app.ActivityThread.main(ActivityThread.java:5254)
	at java.lang.reflect.Method.invoke(Native Method)
	at java.lang.reflect.Method.invoke(Method.java:372)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:903)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:698)
```

See possibly related https://github.com/google/gson/issues/1354

@LukasPaczos and I have found that instrumentation tests will actually pass fine, and it's only the robo testing that fails with this error.

### CODE REVIEW

**Comment by langsmith** [comment]
👀

**Comment by danesfeder** [comment]
@devotaaabel @Guardiola31337 this is ready for review - unblocks the failing robo tests

File: app/src/androidTest/java/com/mapbox/services/android/navigation/ui/v5/NavigationMapRouteTest.java [hunk 1]
```diff
@@ -1,200 +0,0 @@
...
-
-import timber.log.Timber;
-
-import static android.support.test.espresso.Espresso.onView;
-import static android.support.test.espresso.assertion.ViewAssertions.matches;
-import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
-import static android.support.test.espresso.matcher.ViewMatchers.withId;
-import static junit.framework.Assert.assertTrue;
-
-@RunWith(AndroidJUnit4.class)
```
Comment (devotaaabel) [inline app/src/androidTest/java/com/mapbox/services/android/navigation/ui/v5/NavigationMapRouteTest.java]: Maybe you could just remove this line and it won't run? So you don't have to remove the entire file. (I'm not sure if that'll work- I would double check)

File: app/src/androidTest/java/com/mapbox/services/android/navigation/testapp/NavigationViewActivityTest.java [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+  private void enableTestMode() {
+    Context context = mActivityTestRule.getActivity();
+    SharedPreferences preferences = PreferenceManager.getDefaultSharedPreferences(context);
+    SharedPreferences.Editor editor = preferences.edit();
+    editor.putBoolean(context.getString(R.string.test_mode_key), true);
+    editor.apply();
+  }
+
+  private void sleep(long millis) {
```
Comment (devotaaabel) [inline app/src/androidTest/java/com/mapbox/services/android/navigation/testapp/NavigationViewActivityTest.java]: maybe have a `BaseTest` that has the `setup` and `sleep` methods? And anything else used in multiple tests

**Comment by danesfeder** [comment]
Closing this in favor of #1179 and the robo testing being fixed.

**Pull request closed without merging**